### PR TITLE
fix signature hints, show signature on hover

### PR DESCRIPTION
**Pull request by pgrivachev**
- Fixed some cases that didn't work before
- Added unit/e2e tests
- Added the ability to show signature on hover
image

### CODE REVIEW

**Review by fivetran-dmitryboykov — COMMENTED** [review]
Looks good. Could you please answer the questions?

File: server/src/utils/TextUtils.ts [hunk 1]
```diff
@@ -202,3 +202,37 @@ function createWordRegExp(allowInWords = ''): RegExp {
   source += '\\s]+)';
   return new RegExp(source, 'g');
 }
+
+export function getTextRangeBeforeBracket(text: string, cursorPosition: Position): Range {
```
[the remark this answers is not in the document]
Comment (pgrivachev) [reply server/src/utils/TextUtils.ts:206]: Refactored a bit, thanks.

File: server/src/SignatureHelpProvider.ts [hunk 2]
```diff
@@ -12,14 +12,12 @@ export interface SignatureInfo {
...
-  signatureInformations = new Map<string, SignatureInformation[]>();
-
-  onSignatureHelp(params: SignatureHelpParams, text: string): SignatureHelp | undefined {
+  onSignatureHelp(text: string): SignatureHelp | undefined {
     const index = HelpProviderWords.findIndex(w => w.name === text);
     if (index !== -1) {
       return {
         signatures: HelpProviderWords[index].signatures.map<SignatureInformation>(s => ({
-          label: s.signature,
+          label: `${s.signature}`,
```
[the remark this answers is not in the document]
Comment (pgrivachev) [reply server/src/SignatureHelpProvider.ts]: Yes, it is not needed.

File: server/src/DbtTextDocument.ts [hunk 3]
```diff
@@ -331,8 +333,8 @@ export class DbtTextDocument {
   }
 
   onSignatureHelp(params: SignatureHelpParams): SignatureHelp | undefined {
-    const text = this.rawDocument.getText(this.getTextRangeBeforeBracket(params.position));
-    return this.signatureHelpProvider.onSignatureHelp(params, text);
+    const text = this.rawDocument.getText(getTextRangeBeforeBracket(this.rawDocument.getText(), params.position));
```
Comment (fivetran-dmitryboykov) [inline server/src/DbtTextDocument.ts:336]: Why do we need `getTextRangeBeforeBracket`? It seems that we can use `getWordRangeAtPosition`.

**Review by fivetran-dmitryboykov — COMMENTED** [review]
Good. Just one question.

File: server/src/DbtTextDocument.ts (hunk 3 above)
Comment (pgrivachev) [reply server/src/DbtTextDocument.ts:336]: Now I use `getWordRangeAtPosition` internally in https://github.com/fivetran/dbt-language-server/pull/99/files#diff-7ee293d1f12e6e9036c6bf8d2558a887989691cf80f7f394c4d61dca27594e22R206. But in this method we need find Bracket depending on context. For `foo(bar(abc), d) ` this methos should return foo if cursor located in position of `d`. I'm not sure that we can solve this only by using `getWordRangeAtPosition`, or we should have very complex regexp for it.

Comment (fivetran-dmitryboykov) [reply server/src/DbtTextDocument.ts:336]: Seems reasonable.

**Review by fivetran-dmitryboykov — APPROVED** [review]

**Pull request merged**